core/math/transform_2d.cpp

```diff
@@ -172,6 +172,10 @@ bool Transform2D::is_finite() const {
 	return columns[0].is_finite() && columns[1].is_finite() && columns[2].is_finite();
 }
 
+bool Transform2D::is_invertible() const {
+	return basis_determinant() != 0;
+}
```

Comment:
I noticed, that the approach with `is_invertible()` before calculating `affine_inverse()` has the drawback, that the basis determinant is computed twice.

A different approach would be 
``` diff
diff --git a/core/math/transform_2d.cpp b/core/math/transform_2d.cpp
index 548a82d254..55a12249c7 100644
--- a/core/math/transform_2d.cpp
+++ b/core/math/transform_2d.cpp
@@ -47,9 +47,6 @@ Transform2D Transform2D::inverse() const {
 
 void Transform2D::affine_invert() {
        real_t det = basis_determinant();
-#ifdef MATH_CHECKS
-       ERR_FAIL_COND(det == 0);
-#endif
        real_t idet = 1.0f / det;
 
        SWAP(columns[0][0], columns[1][1]);
```
and to test afterwards:
``` c++
Transform2D xform = x.affine_inverse();
if (xform.is_finite()) { ...
```